### PR TITLE
Backport of several IP blocks and email rate limiting from private repo

### PR DESCRIPTION
**Pull request by seanmonstar**

### CODE REVIEW

**Comment by vladikoff** [comment]
Need a rebase and travis fix

**Comment by seanmonstar** [comment]
rebased and green!

File: test/remote/block_ip_tests.js [hunk 1]
```diff
@@ -47,74 +40,77 @@ test(
...
+              function (err, req, res, obj) {
+                t.equal(res.statusCode, 200, 'check worked')
+                t.equal(obj.block, true, 'request was blocked')
+                t.end()
+              }
+            )
+          }
+        )
+      }
+    )
```
Comment (rfk) [inline test/remote/block_ip_tests.js]: Hrm, looks like this undid some of @larissagaulia's fine work on promisifying the tests from https://github.com/mozilla/fxa-customs-server/commit/6fadc52bfabfc3d08dbbdf3f4a42bf67ec9b54cf, was this a bad merge of some sort @seanmonstar?

Comment (seanmonstar) [inline test/remote/block_ip_tests.js]: Ah! I though it had said `git checkout --theirs`... *grumble*

**Comment by seanmonstar** [comment]
@rfk fixed the bad merge

**Comment by rfk** [comment]
@seanmonstar test bustage?

This issue is strong encouragement for us to work on getting out of that private repo!

**Comment by seanmonstar** [comment]
Hm, seems the test that was reverted in the 3rd commit is failing. I'll try killing that commit, making the test now check that the request **was blocked**.

**Comment by seanmonstar** [comment]
@rfk travis says green

**Comment by rfk** [comment]
Hmm, looks like the public repo is also missing a refactor of how allowedIPs is handled, which is why the test behaves differently here vs in the private repo :-(

Anyways, that's a separate problem, r+ on this one

**Pull request merged**